lib/nostrum/cache/guild/guild_server.ex

```diff
@@ -442,4 +447,14 @@ defmodule Nostrum.Cache.Guild.GuildServer do
     {:reply, {guild_id, old_emojis, emojis}, %{state | emojis: emojis}}
   end
 
+  def handle_call({:update, :voice_state, guild_id, voice_state}, _from, state) do
+    old_voice_states = state.voice_states
+    new_voice_states =
+      [
+        voice_state |
+        state.voice_states
+        |> Enum.reject(fn(map) -> map.user_id == voice_state.user_id end)
+      ]
+    {:reply, {guild_id, old_voice_states, new_voice_states}, %{state | voice_states: new_voice_states}}
```

Comment:
Also,

> Since the number of voice states will be fewer than other lists (such as presences), it may not be a big deal to dispatch whole lists.

The number of voice states might become the same count as the presences list, because currently we do not delete the state from the list, when the state for one user becomes `channel_id: nil`. So the max count of voice states equals the number of the guild members (including kicked members).